wolfcrypt/src/coding.c

```diff
@@ -236,17 +231,104 @@ int Base64_Decode(const byte* in, word32 inLen, byte* out, word32* outLen)
             WOLFSSL_MSG("Bad Base64 Decode data, too big");
             return ASN_INPUT_E;
         }
-#endif
 
         if (i + 1 + !pad3 + !pad4 > *outLen) {
             WOLFSSL_MSG("Bad Base64 Decode out buffer, too small");
             return BAD_FUNC_ARG;
         }
 
-        e1 = Base64_Char2Val(e1);
-        e2 = Base64_Char2Val(e2);
-        e3 = (byte)((e3 == PAD) ? 0 : Base64_Char2Val(e3));
-        e4 = (byte)((e4 == PAD) ? 0 : Base64_Char2Val(e4));
+        e1 = Base64_Char2Val_by_table(e1);
+        e2 = Base64_Char2Val_by_table(e2);
+        e3 = (byte)((e3 == PAD) ? 0 : Base64_Char2Val_by_table(e3));
+        e4 = (byte)((e4 == PAD) ? 0 : Base64_Char2Val_by_table(e4));
+
+        if (e1 == BAD || e2 == BAD || e3 == BAD || e4 == BAD) {
+            WOLFSSL_MSG("Bad Base64 Decode bad character");
+            return ASN_INPUT_E;
+        }
+
+        b1 = (byte)((e1 << 2) | (e2 >> 4));
+        b2 = (byte)(((e2 & 0xF) << 4) | (e3 >> 2));
+        b3 = (byte)(((e3 & 0x3) << 6) | e4);
+
+        out[i++] = b1;
+        if (!pad3)
+            out[i++] = b2;
+        if (!pad4)
+            out[i++] = b3;
+        else
+            break;
+    }
+/* If the output buffer has a room for an extra byte, add a null terminator */
```

Comment:
Bad format.

Reply:
fixed.  piggybacked on attempted fixes to the broken workflows.